File: ios/MobilePassSDK/Model/QRCodeProcessResult.swift
//
//  QRCodeProcessResult.swift
//  MobilePassSDK
//
//  Created by Mobile Pass SDK
//

import Foundation

public struct QRCodeProcessResult {
    public var isValid: Bool
    public var errorType: QRCodeErrorType?
    
    init(isValid: Bool, errorType: QRCodeErrorType? = nil) {
        self.isValid = isValid
        self.errorType = errorType
    }
}

File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/QRCodeProcessResult.java
package com.armongate.mobilepasssdk.model;

import androidx.annotation.Nullable;

import com.armongate.mobilepasssdk.enums.QRCodeErrorType;

/**
 * Result of QR code processing
 */
public class QRCodeProcessResult {
    private final boolean isValid;
    private final QRCodeErrorType errorType;

    public QRCodeProcessResult(boolean isValid, @Nullable QRCodeErrorType errorType) {
        this.isValid = isValid;
        this.errorType = errorType;
    }

    public boolean isValid() {
        return isValid;
    }

    @Nullable
    public QRCodeErrorType getErrorType() {
        return errorType;
    }
}

